src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1136,6 +1176,49 @@ public SolutionCompilationState WithOptions(SolutionOptionSet options)
             this.SolutionState.WithOptions(options));
     }
 
+    public SolutionCompilationState WithSourceGeneratorExecutionVersions(
+        SourceGeneratorExecutionVersionMap sourceGeneratorExecutionVersions, CancellationToken cancellationToken)
```

Comment:
>CancellationToken cancellationToken [](http://example.com/codeflow?start=77&length=35)

Seems a little odd to plumb a CancellationToken through to this method. Is there something happening in the loop that is more expensive than I'm expecting?

Reply:
it has the loop, and it calls another helper (WithCreationPolicy) that takes a canclelation token.  for hygiene i want to pass this along so that if ever needed or expensive, it can be canceled.

Reply:
Ahh, didn't notice that WithCreationPolicy took in the cancellationToken